### ISSUE TITLE
Crash in browser while selecting and de-selecting multiple images

### ISSUE THREAD

**Issue by mattiaverga** [Beep6581/RawTherapee#5701]
Rawtherapee 5.8
RPMs from Fedora 31 repository

Download test files from https://eu.storagemadeeasy.com/folders/53cb0b4760077e3191e65f008dcf649c (password "rawtherapee")
From Rawtherapee browser open the folder where you put those files.
Select all three files with shift-left click.
Left click again on one of those files to de-select the others.
Rawtherapee crashes, from console output:
```
/usr/include/c++/9/bits/stl_vector.h:1042: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = Crop::CropRatio; _Alloc = std::allocator<Crop::CropRatio>; std::vector<_Tp, _Alloc>::reference = Crop::CropRatio&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__builtin_expect(__n < this->size(), true)' failed.
```
The same happen if you select all three files (or two of them) and try to apply a color label.

If you delete the pp3 files, this doesn't happen. Those files were processed with an older RT version (I don't remember exactly what).

**Comment by mattiaverga** [comment]
A more detailed backtrace can be found at https://bugzilla.redhat.com/show_bug.cgi?id=1820907

**Comment by Floessie** [comment]
@mattiaverga Cannot reproduce with the files you provided, but I'm not using the same hardening flags as you on Fedora. Nevertheless I tried to work around the crash in `crop.cc`, maybe you could give current `dev` a try...

**Comment by mattiaverga** [comment]
For a quicker test I have applied your patch on top of 5.8 stable and I confirm the crash is fixed.
I will ask the original reporter on RH Bugzilla to try the patched release to have a second confirmation.
Thank you!

**Comment by mattiaverga** [comment]
The OP has confirmed the patch has fixed the issue.

**Comment by Floessie** [comment]
Okay. Closing as fixed with 15db8cdb57161d57af8307bf85aeda02ea69ddde.
